once_test.py

```diff
@@ -519,6 +536,40 @@ def once_fn(self):
         self.assertEqual(b.once_fn(), 1)
         self.assertEqual(b.once_fn(), 1)
 
+    def test_once_per_class_parallel(self):
+        class _CallOnceClass(Counter):
+            @once.once_per_class
+            def once_fn(self):
+                return self.get_incremented()
+
+        once_obj = _CallOnceClass()
+
+        @execute_with_barrier(n_workers=_N_WORKERS)
+        def execute(_):
+            return once_obj.once_fn()
+
+        with concurrent.futures.ThreadPoolExecutor(max_workers=_N_WORKERS) as executor:
+            results = list(executor.map(execute, range(_N_WORKERS * 4)))
+        self.assertEqual(min(results), 1)
+        self.assertEqual(max(results), 1)
+
+    def test_once_per_class_per_thread(self):
+        class _CallOnceClass(Counter):
+            @once.once_per_class.with_options(per_thread=True)
+            def once_fn(self):
+                return self.get_incremented()
+
+        once_obj = _CallOnceClass()
+
+        @execute_with_barrier(n_workers=_N_WORKERS)
```

Comment:
For both this and above, moving the barrier inside the func would actually test that they're running in parallel. I think you would
```python
@once.once_per_class.with_options(per_thread=True)
@execute_with_barrier(n_workers=_N_WORKERS)  # Requires integer number of _N_WORKERS threads.
def once_fn(self):
    return self.get_incremented()
```

Reply:
Done

Reply:
You can get rid of execute and just pass lambda _: once_obj.once_fn directly to executor if you want. Fine if you want to keep it as a closure for consistency with the other tests, but let's get rid of all the superfluous barriers and keep just the load bearing ones.